src/lua/uri.lua

```diff
@@ -169,8 +171,52 @@ local function format(uri, write_password)
     return str
 end
 
+local function encode_kv(key, values)
+    local val = values
+    if type(val) ~= "table" then
+        val = {val}
+    elseif table.getn(values) == 0 then
+        return ("%s"):format(key)
+    end
+    local res
+    for _, v in pairs(val) do
+        local kv = ("%s=%s"):format(key, v)
+        if not res and kv then
+            res = kv
+        elseif kv then
+            res = ("%s&%s"):format(res , kv)
+        end
+    end
```

Comment:
Here we can also just add `'a=b'` strings to an accumulator table defined in `params` and it'll be concatenated into `<...>&<...>&<...>` in params automatically. This way it'll produce less intermediate strings and, what is more important, the code will be simpler.